frontend/src/components/OperationsTableCard/OperationsTableCard.tsx

```diff
@@ -57,17 +58,25 @@ const getSourceDatasetsLink = (
   }${sourceID}?limit=${limit}&offset=${offset}&search=${search}`;
 
 const OperationsTableCard: FunctionComponent<OperationsTableCardProps> = (props) => {
+  const { search } = useLocation();
+  const queryParams = queryString.parse(location.search);
   const [showMyQueries, setShowMyQueries] = useState(props.showMyQueries);
-  const [searchQuery, setSearchQuery] = useState('');
+  const [searchQuery, setSearchQuery] = useState((queryParams.q as string) || '');
   const [info, setInfo] = useState('');
   const [dropDownValues, setDropDownValues] = useState<DropdownItemProps[]>([]);
   const onModalHide = () => setInfo('');
   const { sources } = useContext(SourcesContext);
-
+  const [pageNumber, setPageNumber] = useState(Number(queryParams.page as string) || 1);
+  useEffect(() => {
+    if (Object.entries(queryParams).length === 0) {
+      props.history.push(`${window.location.pathname}?page=1`);
+      setPageNumber(1);
+    }
+    setPageNumber(Number(queryParams.page as string));
+  }, [queryParams]);
   useEffect(() => {
     fetchQueries(showMyQueries);
-  }, []);
-
+  }, [pageNumber, search]);
```

Comment:
Why watch these when you're also watching the query params? Seems redundant ... don't mind it though ... I feel a lot of this is inefficient but you've been on it too long & it needs to be closed.

Reply:
The queryparams are being watched in a separate useEffect